### PR TITLE
Feat: Add Shortcuts Logic to toolbar

### PR DESCRIPTION
**Pull request by aridanemartin**
This PR adds the possibility of using shortcuts for the main buttons in the toolbar: 
- Keys in Windows/Linux will use 'Alt' as the modifier to not interfiere with browser defaults.
- Keys in Mac will use 'Alt' as the modifier to not interfiere with browser defaults.
- The functionality supports Spanish/English keyboard.

Notes
- Accessibility is being considered using describedby referring to the tooltip
- Tooltip won't be shown if the button is disabled.

https://github.com/Lemoncode/mongo-modeler/assets/63003079/ad5e1d05-8a68-40b3-a2d5-23eaf7b86476

### CODE REVIEW

**Comment by aridanemartin** [comment]
Close #322

File: src/pods/toolbar/components/toolbar-button/toolbarButton.component.tsx [hunk 1]
```diff
@@ -1,23 +1,55 @@
 import React from 'react';
+import useShortcut from '../../shortcut/shortcut.hook';
+import { isMacOS } from '@/common/helpers/platform.helpers';
+
+export interface ShortcutOptions {
+  id: string;
+  targetKey: string[];
+  targetKeyLabel: string;
+  description: string;
+}
```
Comment (antonio06) [inline src/pods/toolbar/components/toolbar-button/toolbarButton.component.tsx:10]: i think that its more convenient move it to shortcut.model or somethink like that

File: src/pods/toolbar/shortcut/shortcut.const.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
+export const SHORTCUTS = {
```
Comment (antonio06) [inline src/pods/toolbar/shortcut/shortcut.const.ts:1]: use ShortcutOptions  type here

File: src/pods/toolbar/shortcut/shortcut.hook.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ * @param  {String[]} targetKey The key that will trigger the shortcut
+ * @param  {Function} callback  The function to be called when the shortcut is triggered
+ * @return {void}
+ */
+
+const useShortcut = ({ targetKey, callback }: ShortcutHookProps) => {
+  const handleKeyPress = (event: KeyboardEvent) => {
+    const isAltKeyPressed = event.getModifierState('Alt');
+    const isCtrlKeyPressed = event.getModifierState('Control');
+    console.log('===event.key==>', event.key);
```
Comment (antonio06) [inline src/pods/toolbar/shortcut/shortcut.hook.tsx:22]: remove console.log line

File: src/pods/toolbar/shortcut/shortcut.hook.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    if (
+      (isWindowsOrLinux() && isAltKeyPressed) ||
+      (isMacOS() && isCtrlKeyPressed)
+    ) {
+      if (targetKey.includes(event.key)) {
+        console.log('===event.key==>', event.key);
```
Comment (antonio06) [inline src/pods/toolbar/shortcut/shortcut.hook.tsx:28]: remove console.log line

File: src/pods/toolbar/shortcut/shortcut.const.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
+import { ShortcutOptions } from './shortcut.model';
+
+interface ShortcutList {
```
Comment (antonio06) [inline src/pods/toolbar/shortcut/shortcut.const.ts:3]: this is not a list, is a object. Rename this to Shortcut

**Pull request merged**